src/modules/client/pages/ni/billing/CreditNotes.vue

```diff
@@ -512,23 +560,37 @@ export default {
 
       return payload;
     },
+    formatPayloadWithBillingItems (creditNote) {
+      return {
+        ...pick(creditNote, ['exclTaxesCustomer', 'inclTaxesCustomer']),
+        billingItemList: creditNote.billingItemList.map(bi => omit(bi, 'vat')),
+      };
+    },
     formatPayload (creditNote) {
       let payload = pick(creditNote, ['date', 'customer', 'misc']);
 
       if (this.creditNoteType === SUBSCRIPTION) {
         payload = { ...payload, ...this.formatPayloadWithSubscription(creditNote) };
-      } else payload = { ...payload, ...this.formatPayloadWithLinkedEvents(creditNote) };
+      } else if (this.creditNoteType === EVENTS) {
+        payload = { ...payload, ...this.formatPayloadWithLinkedEvents(creditNote) };
+      } else {
+        payload = { ...payload, ...this.formatPayloadWithBillingItems(creditNote) };
+      }
 
       return pickBy(payload);
     },
     async createNewCreditNote () {
       try {
+        if (this.creditNoteType !== BILLING_ITEMS) this.newCreditNote.billingItemList = [];
+        const hasDuplicateBillingItems = this.newCreditNote.billingItemList.map(bi => bi.billingItem).length
+          === [...new Set(this.newCreditNote.billingItemList.map(bi => bi.billingItem))].length;
```

Comment:
Que penses-tu de creer une constante pour `this.newCreditNote.billingItemList.map(bi => bi.billingItem)`